python/taulabs/telemetry.py

```diff
@@ -323,13 +323,19 @@ def _do_io(self, finish_time):
         if r:
             # Shouldn't throw an exception-- they just told us
             # it was ready for read.
-            chunk = os.read(self.fd, 1024)
-            if chunk == '':
-                raise RuntimeError("stream closed")
+            try:
```

Comment:
I'm a little concerned about this-- worries me it's masking a bug with when things are ready for reading.  EAGAIN after they told us it's ready to read?  How odd.  Maybe I am calling this at inappropriate times.